src/main/haskell/kore/src/Kore/AST/BuildersImpl.hs

```diff
@@ -25,14 +25,15 @@ checking that the sorts are identical where possible, creating a pattern with
 the provided sort otherwise.
 -}
 fillCheckSorts
-    ::  ( Show level
-        , Show (variable level)
-        , Show (Pattern level domain variable child)
-        , child ~ PureMLPattern level domain variable
+    ::  ( Functor dom
+        , Show lvl
+        , Show (var lvl)
+        , Show (Pattern lvl dom var child)
+        , child ~ PurePattern lvl dom var ()
         )
-    => [Sort level]
-    -> [PurePatternStub level domain variable]
-    -> [PureMLPattern level domain variable]
+    => [Sort lvl]
```

Comment:
I can't think of a single other word `lvl` could be short for..

Reply:
This does not answer any of the two questions asked before, so I'm not sure what point you were trying to make.

Still, I'll attempt to add a few words about abbreviations, since I don't agree with the implicit ideas behind what you said.

First, I know what `lvl` is, and everyone in our project does. Now, reading usually works by pattern-matching on words or larger entitites, and pattern-matching will work better with full words, the kind we all have been reading for tens of years. So our reading will be a tiny bit slower, but we can get used to different patterns, so it may not be a big issue. But the problem is that we shouldn't be writing code only for ourselves, both because our code is open-source and because we might hire other coders.

Mind you, this is not a huge issue, but why would you want to make everyone's reading more inefficient? Especially since, if our project is successful, it is likely that many other people will read parts of it.

Second, the "can't think of another word" part is missing the point. The actual problem is finding the right word. To go from `level` to `level` is trivial. To go from `lvl` to `level` is a bit harder. To go from `dom` to `domain` is even harder.

Third, the "can't think of another word" part relies on people being used too much to our project and lacking imagination. I mean, why not `lvl` -> `lovely`? I bet I could find lots of examples that roughly fit `l([aeiou])*v([aeiou])*l([aeiou])*`. And `dom` is a prefix, so it is even more ambiguous.